machine-learning/text-count-vectorizer.ipynb

```diff
@@ -0,0 +1,212 @@
+    "lengths = [0] + list(lengths) + [None]\n",
+    "\n",
+    "slices = [slice(a, b) for a, b in\n",
+    "          toolz.sliding_window(2, lengths)]\n",
+    "# Notice the persist here! More details later.\n",
+    "documents = db.from_delayed([load_news(x) for x in slices]).persist()\n",
+    "documents"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask_ml.feature_extraction.text"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vectorizer = dask_ml.feature_extraction.text.CountVectorizer()\n",
+    "%time result = vectorizer.fit_transform(documents)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The call to `fit_transform` did some work to discover the *vocabulary*, a mapping from terms in the documents to positions in the transformed result array."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "list(vectorizer.vocabulary_.items())[:5]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Speaking of the result, it's a Dask `Array` backed by `scipy.sparse.csr_matrix` objects. We can bring it back to the client with `.compute()`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "local_result = result.compute()\n",
+    "local_result[:5].toarray()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that we persisted `documents` earlier. If possible, persisting the input documents is preferable to avoid making two passes over the data. One to discover the vocabulary and a second to transform. If the dataset is larger than (distributed) memory, then two passes will be necessary."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##  A note on vocabularies\n",
+    "\n",
+    "You can also provide a vocabulary ahead of time, which avoids the need for making two passes over the data. This makes operations like `vectorizer.transform` instantaneous, since no vocabulary needs to be discovered. However, vocabularies can become quite large. Consider persisting your data ahead of time to avoid bloating the size of the `CountVectorizer` object. Dask-ML's `CountVectorizer` works just fine when the `vocabulary` is a pointer to a piece of data on the cluster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vocabulary = vectorizer.vocabulary_\n",
+    "remote_vocabulary, = client.scatter([vocabulary], broadcast=True)\n",
+    "\n",
+    "vectorizer2 = dask_ml.feature_extraction.text.CountVectorizer(\n",
+    "    vocabulary=remote_vocabulary\n",
+    ")"
```

Comment:
Is there a well defined vocabulary that we can use somewhere?  Maybe in `nltk`?  I'm concerned that people will see this, and think that they should copy the vocabulary off of one CountVectorizer and then pass it to another.  

Also, do we need the scatter? Can you verify that if vocabulary is included directly in the `vocabulary=` keyword argument that it will occupy only a single task, and not be in many of them?

Reply:
I'm not sure about nltk, but probably not worth adding it to the environment just for this example. I noted that you'd probably get this from an external source in practice.

I think that as of https://github.com/dask/dask-ml/pull/719, the answer to your question about user-provided vocabulary being in one task is "yes". But that change hasn't been released yet.